build/media_source/plg_system_shortcut/keyselect.es6.js

```diff
@@ -0,0 +1,81 @@
+class JoomlaShortcutModal {
+	constructor() {
+		
+		if (!Joomla) {
+			throw new Error('Joomla API is not properly initialised');
+		}
+		// Bindings
+    	this.setAttributes = this.setAttributes.bind(this);
+		this.buildKeySelectModal = this.buildKeySelectModal.bind(this);
+
+		document.addEventListener("DOMContentLoaded",this.buildKeySelectModal);
+		
+		var keySelectBtns = document.getElementsByClassName("keySelectBtn");
+		for(var x = 0; x < keySelectBtns.length; x++) {
+			keySelectBtns[x].addEventListener("click",this.handleClickEvent,false);
+		}
+		
+	}
+	
+	buildKeySelectModal() {
+		
+		
+		const modal_div = document.createElement("div");
+		this.setAttributes(modal_div, {"class": "modal fade", "id": "keySelectModal", "tabindex": "-1", "role": "dialog", "aria-labelledby": "keySelectModalLabel", "aria-hidden": "true"});
```

Comment:
why we are using javascript to render modal?  why not in `PHP`  🧐

Reply:
because of the Joomla! plugin limitation to render HTML with in admin section, this is the only option I think of right now.